hal/spi_api.h

```diff
@@ -19,220 +18,331 @@
 #ifndef MBED_SPI_API_H
 #define MBED_SPI_API_H
 
-#include "device.h"
-#include "hal/dma_api.h"
-#include "hal/buffer.h"
-
 #if DEVICE_SPI
+/**
+ * \defgroup hal_new_spi Serial peripheral interface HAL API.
+ * Low level interface to the serial peripheral interface of a target.
+ *
+ * A SPI peripheral might be used by multiple spi_t as it only represents a communication
+ * channel towards a slave (or a master).
+ * Two SPI instances using the same peripheral can be identified by their SS pin.
+ *
+ * # Defined behaviour
```

Comment:
Can we copy the behaviours into each of the functions descriptions.

Reply:
I tried to slightly rephrase them in each header to make them more readable.